tests/precompile/solidity/suites.go

```diff
@@ -6,13 +6,106 @@ package solidity
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
 	"time"
 
+	"github.com/ava-labs/avalanchego/api/health"
 	"github.com/ava-labs/subnet-evm/tests/utils"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/go-cmd/cmd"
 	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
 )
 
-var _ = ginkgo.Describe("[Precompiles]", ginkgo.Ordered, func() {
+// Timeout to boot the AvalancheGo node
+var bootAvalancheNodeTimeout = 5 * time.Minute
+
+// Timeout for the health API to check the AvalancheGo is ready
+var healthCheckTimeout = 5 * time.Second
+
+// At boot time subnets are created, one for each test suite. This global
+// variable has all the subnets IDs that can be used.
+//
+// One process creates the AvalancheGo node and all the subnets, and these
+// subnets IDs are passed to all other processes and stored in this global
+// variable
+var subnetsIds map[string]string
+
+// Keep track of the AvalancheGo external bash script, it is null for most
+// processes except the first process that starts AvalancheGo
+var startCmd *cmd.Cmd
+
+// Out test suite runs in a separated processes, ginkgo has
+// SynchronizedBeforeSuite() which is promised to run once, and its output is
+// passed over to each worker.
+//
+// In here an AvalancheGo node instance is started, and subnets are created for
+// each test case. Each test case has its own subnet, therefore all tests can
+// run in parallel without any issue.
+//
+// This function also compiles all the solidity contracts
+var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
+	ctx, cancel := context.WithTimeout(context.Background(), bootAvalancheNodeTimeout)
+	defer cancel()
+
+	wd, err := os.Getwd()
+	gomega.Expect(err).Should(gomega.BeNil())
+	log.Info("Starting AvalancheGo node", "wd", wd)
+	cmd, err := utils.RunCommand("./scripts/run.sh")
+	startCmd = cmd
+	gomega.Expect(err).Should(gomega.BeNil())
+
+	// Assumes that startCmd will launch a node with HTTP Port at [utils.DefaultLocalNodeURI]
+	healthClient := health.NewClient(utils.DefaultLocalNodeURI)
+	healthy, err := health.AwaitReady(ctx, healthClient, healthCheckTimeout, nil)
+	gomega.Expect(err).Should(gomega.BeNil())
+	gomega.Expect(healthy).Should(gomega.BeTrue())
+	log.Info("AvalancheGo node is healthy")
+
+	subnets := make(map[string]string)
+	files, err := filepath.Glob("./tests/precompile/genesis/*.json")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	for _, file := range files {
+		basename := filepath.Base(file)
+		index := basename[:len(basename)-5]
+		subnets[index] = utils.CreateNewSubnet(ctx, file)
+	}
+
+	subnetsBytes, err := json.Marshal(subnets)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	compileContracts()
+	return subnetsBytes
+}, func(data []byte) {
+	err := json.Unmarshal(data, &subnetsIds)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+})
```

Comment:
This is really nice. Putting this right here, this code is really clear that the `SynchronizedBeforeSuite` function generates this result and then the next function running in every process will unmarshal that result to ensure that this variable is set correctly in every process ginkgo is running.